### PR TITLE
Fix metricbeat magefile to generate metricbeat/include/fields.go and run make update

### PR DESCRIPTION
**Pull request by MichaelKatsoulis**
## What does this PR do?

This PR updates metricbeat magefile in order to generate metricbeat/include/fields/fields.go. 

## Why is it important?

The fields.go file was not being updated after `make update`.  In case a field was updated in fields which where included in metricbeat  this was  leading to tests failing.

For example updating an existing field in [libbeat/processors/add_kubernetes_metadata/_meta/fields.yml](https://github.com/elastic/beats/blob/master/libbeat/processors/add_kubernetes_metadata/_meta/fields.yml), then [metricbeat/include/fields/fields.go](https://github.com/elastic/beats/blob/master/metricbeat/include/fields/fields.go) was not getting updated.

When tests run they get the available fields from [FieldsRegistry](https://github.com/elastic/beats/blob/86004dd4ba40b40e3e88344e7e4c0e47670886d7/libbeat/asset/registry.go#L32)
which gets filled by the fields.go files. 

So as far as the unit [tests](https://github.com/elastic/beats/blob/86004dd4ba40b40e3e88344e7e4c0e47670886d7/metricbeat/mb/testing/testdata.go#L320) are concerned the fields where not updated, thus leading to failures.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

After making an update in [libbeat/processors/add_kubernetes_metadata/_meta/fields.yml](https://github.com/elastic/beats/blob/master/libbeat/processors/add_kubernetes_metadata/_meta/fields.yml),
run `make update` in metricbeat and notice the metricbeat/include/fields/fields.go getting updated.

## Related issues

- Closes https://github.com/elastic/beats/issues/28021

### CODE REVIEW

File: metricbeat/include/fields/fields.go [hunk 1]
```diff
@@ -24,13 +24,13 @@ import (
 )
 
 func init() {
-	if err := asset.SetFields("metricbeat", "../libbeat/fields.yml", asset.LibbeatFieldsPri, AssetLibbeatFieldsYml); err != nil {
+	if err := asset.SetFields("metricbeat", "fields.yml", asset.BeatFieldsPri, AssetFieldsYml); err != nil {
```
Comment (ruflin) [inline metricbeat/include/fields/fields.go:27]: I'm not sure if this is correct. Previously it was generated based of the libbeat fields.yml which only contains the libbeat fields. Now it seems to be changed to the metricbeat fields.yml which likely also contains more fields from metricbeat itself. Do we have duplicated fields now or are by chance missing some?

File: metricbeat/magefile.go [hunk 2]
```diff
@@ -133,7 +133,17 @@ func MockedTests(ctx context.Context) error {
 
 // Fields generates a fields.yml and fields.go for each module.
 func Fields() {
-	mg.Deps(fieldsYML, moduleFieldsGo)
+	mg.Deps(libbeatAndMetricbeatCommonFieldsGo, moduleFieldsGo)
+	mg.Deps(fieldsYML)
```
Comment (ruflin) [inline metricbeat/magefile.go:137]: Do we need the `fieldsYML` in addition. Shouldn't the module fields and common fields be enough as that is what is fed into the fieldsYML?

File: metricbeat/module/kubernetes/state_pod/_meta/fields.yml [hunk 3]
```diff
@@ -4,10 +4,6 @@
     kubernetes pod metrics
   release: ga
   fields:
-    - name: ip
```
Comment (ruflin) [inline metricbeat/module/kubernetes/state_pod/_meta/fields.yml:7]: How is this change related?

**Review by ruflin — COMMENTED** [review]
Thanks for taking this on. Left some questions. I remember in Metricbeat it is a bit of juggling around the fields.yml generation based on the order these are generated and updated. 

One of the main questions is if we build off the fields.go out of 1 complete `fields.yml` or we generated many small `fields.go` out of the local `fields.yml`. The global `fields.go` historically has generated many pull request conflicts and that is why we tried to get rid of it /  reduce it so that in case someone changes a processor or a module the PR only touches these parts. 

@andrewkroh I would appreciate your review / feedback on this PR as maybe it is fresher in your mind on what we did here also with the move to mage.

File: metricbeat/include/fields/fields.go (hunk 1 above)
Comment (MichaelKatsoulis) [reply metricbeat/include/fields/fields.go:27]: I checked for duplicates and we didn't have any. I will try to understand why. 
But my first approach was to use `../libbeat/fields.yml`  for that fields.go.  It should be the best solution, but there as some other fields missing which are the [common fields](https://github.com/elastic/beats/blob/master/metricbeat/_meta/fields.common.yml).
We would  need a fields.go for them also and then it starts getting more complicated.

**Comment by MichaelKatsoulis** [comment]
@ruflin I will try to explain it the way I see it. In function 
```
// libbeatAndMetricbeatCommonFieldsGo generates a fields.go containing both
// libbeat and metricbeat's common fields.
func libbeatAndMetricbeatCommonFieldsGo() error {
	if err := devtools.GenerateFieldsYAML(); err != nil {
		return err
	}
	return devtools.GenerateMetricbeatAllInOneFieldsGo()
}
```
1. a fields.yaml file is generated in metricbeat folder which does not contain the module fields under each module/_meta/fields.yaml and module/metricset/_meta/fields.yml.  It includes all the fields from libbeat and the common fields.
2. By that fields.yaml, include/fields/fields.go is created which contains the same fields of step 1

After this function is executed by
```
// Fields generates a fields.yml and fields.go for each module.
func Fields() {
	mg.Deps(libbeatAndMetricbeatCommonFieldsGo, moduleFieldsGo)
	mg.Deps(fieldsYML)
}
``` 
3. moduleFieldsGo gets executed which looks inside the modules _meta/fields.yaml and the metricsets/_meta/fields.yml
and generates the fields.go file for each module. This only includes the fields specified in the module. [(code)](https://github.com/elastic/beats/blob/529b9de3468852cf8f5410d179d457b018eb7024/libbeat/generator/fields/module_fields_collector.go#L76). Due to that the final FieldRegistry created by all the fields.go of metricbeat does not contain duplicates.
4. Last step is the creation of the final fields.yml file which contains all the fields (libbeat, common and module fields)
```
func fieldsYML() error {
	return devtools.GenerateFieldsYAML("module")
}
```
This file overrides the one created in step 1
If this file is needed or not I am not sure, but it is something that used to take place before and happens in all beats. Actually I used that to update my metricbeat index mapping in elastic to test a new field I added by running `./metricbeat setup --index-management -E output.logstash.enabled=false -E 'output.elasticsearch.hosts=["https://elastic:K84iwSlE3zBADu9V0YWjqSW3@new-deploy.es.europe-west1.gcp.cloud.es.io:9243"]' -E setup.template.fields="./fields.yml" `

**Review by sayden — COMMENTED** [review]
Let's see CI too and I also think that fieldsYML is not necessary

File: metricbeat/module/kubernetes/fields.go [hunk 4]
```diff
@@ -32,5 +32,5 @@ func init() {
...
...1hwft38er3yTSf39jlrzkR6sGTHhFyT7tZdniMen7JR8BRp9PEmRpg+jXEZCBxXcrPg3YsOzPLyxW11+eIGqNolVqvVsXcuYqKbl5moStYWSmB1FV5xc+G96qPtlm7Mq2qzZmVV7SUi1bUtWFtiQ/UXubykErVW3dKt+cfzVaYdj+vZStJGYGP3uiP2UkLbAi17AtWc0P1Bb58NOBQ472gKje9hmYKCthQ3RZYdKm6D0rTuo+lznz8LJnG0pcWiGWVxWa5c/LbE+r8a61DReFdKUxAYDuZUCFJ0ucM81RuUgPSbUMOtOJFAe6DeuxWK3rEs7BGamaO+eoV+V0P9XY31dzXY3z37h2PgR4zPnLcqURrzw3meERBIsn7EGP6nP8JUywFJYiU8SmrPfu/mrsQRyGdkhZDAfU74CB4fqQROcYY+3tQmX47fzRK+mi/MClKrkVXE0IdPd/4pULM8fpg9hp7YImM4Xd/jDNNkllh/YjhF35d0aoPyMJ0zxauB9WjUNQB0y1VoPMdENAUf+oqBCtnm2ETF5h8uOuEyM28hl1NUmoZaDFtfsKNL2BRZPMe+ohjNsw8JYShZ4y5UrUVS1/+hS1AbtNkH78oRdL2/E4QaLeHVPtRR0cbC/qnVIrdyT1s+n0+I6BnCjl6N/ViAzxZ/DJlgEx0sbYRWHHJ8RdHCtlhboAX2ZdhgZXkjgHXyrt2067wl2c7CPruf10Lj9/Zyzh6JIMxXuDnhcKmh1Hh9NgrfmYE+ulk77k5PCgw0lfIGtml9eaB4TxKsAuZydytPMNxHXeU5yT3RWc9Zaf+fWWoux6agXw5rZEPoFmGaopJLfH+kpfYBr0Q/bB7L+s0r6dY7jlG8EkdDp0macLxuXLdg8DqEp340/U283ZwwZ+HVMPEhBjaT3uWHPpvgU4tofNMqVEr7mnEoRU4x9bRa7KB8Kc9YL1QEdX6muAX69T8rent3N04U5WOsr//t2d96r84OSCbHW1jwRdPmMuDoV1ZPhmj4ndWopWfterNnc9D7UrEqzZzcNvGuETgrzYaIhgijFzkLfyAZlI6peVE8XFSKJh0jv1IRNff+B2Xk6HeBXrt4tDfllYw9iJf2QP9N6zn+ssBCR4teBRPK0kAh+BwVe2wHRXV/vxhleQZhrawcwrFJDDA/cIAxYNy9XGOjMf5rCedfAQAA//8fx9za"
```
[the remark this answers is not in the document]
Comment (MichaelKatsoulis) [reply metricbeat/module/kubernetes/fields.go:35]: If I don't touch this file and the removal of pod.ip in this PR the ci will fail due to fields duplication. The only way would be maybe to not touch also the `include/fields/fields.go` (generated from make update). Then in a separate pr remove the pod.ip and update the fields.go files after this pr is merged. It gets a bit complicated.

**Review by ruflin — APPROVED** [review]
@MichaelKatsoulis Thanks for the detailed explanation. The final fields.yml is (was?) need as it was also packaged with the beat itself as a user was able to modify it and potentially link to the fields.yml to have the template created based on it. But later this should have been replaced by append_fields. If we still support to reference a fields.yml, likely something we should remove in 8.0 (@andresrc ). 

I remember the overwrite of the fields.yml caused some issues but at the same time is intended. @andrewkroh You remember by chance?

Overall LGTM and it sounds like we are in better shape then before. One thing that would be interesting for review is a diff of the `fields.yml` before and after the change. I remember in the past we had checked in the `fields.yml` for this purpose but then got rid of it because of the conflicts.

**Comment by andrewkroh** [comment]
This reminds me of some ideas I had so I put them into an issue at https://github.com/elastic/beats/issues/28138. It touches on some of your points @ruflin. We don't really need the fields.yml that goes into packages, but we do need one that has all fields for executing python tests that check whether all fields in events are documented.

**Comment by MichaelKatsoulis** [comment]
> One thing that would be interesting for review is a diff of the fields.yml before and after the change

@ruflin I diffed the two files. The only difference is that after the change `Kubernetes pod IP` has been deleted from `kubernetes pod metrics` (fields of state_pod metricset)
```
-            - name: ip
-              type: ip
-              description: >
-                Kubernetes pod IP
```
In old fields.yaml before the change it was present two times.
1 Under the libbeat originated fields
2 Under the state_pod fields.

In the new fields.yml it is just one time under the libbeat fields.
I think we are safe.

**Comment by ruflin** [comment]
Lets move forward with this. @andrewkroh My understanding is that your issue is a potential follow.
